packages/chevrotain/src/scan/lexer.ts

```diff
@@ -871,34 +834,38 @@ export function performRuntimeChecks(
     })
   }
 
-  if (has(lexerDefinition, MODES)) {
-    forEach(lexerDefinition.modes, (currModeValue, currModeName) => {
-      forEach(currModeValue, (currTokType, currIdx) => {
-        if (isUndefined(currTokType)) {
-          errors.push({
-            message:
-              `A Lexer cannot be initialized using an undefined Token Type. Mode:` +
-              `<${currModeName}> at index: <${currIdx}>\n`,
-            type: LexerDefinitionErrorType.LEXER_DEFINITION_CANNOT_CONTAIN_UNDEFINED
-          })
-        } else if (has(currTokType, "LONGER_ALT")) {
-          const longerAlt = isArray(currTokType.LONGER_ALT)
-            ? currTokType.LONGER_ALT
-            : [currTokType.LONGER_ALT]
-          forEach(longerAlt, (currLongerAlt) => {
-            if (
-              !isUndefined(currLongerAlt) &&
-              !includes(currModeValue, currLongerAlt)
-            ) {
-              errors.push({
-                message: `A MultiMode Lexer cannot be initialized with a longer_alt <${currLongerAlt.name}> on token <${currTokType.name}> outside of mode <${currModeName}>\n`,
-                type: LexerDefinitionErrorType.MULTI_MODE_LEXER_LONGER_ALT_NOT_IN_CURRENT_MODE
-              })
-            }
-          })
-        }
-      })
-    })
+  if (lexerDefinition.hasOwnProperty(MODES)) {
+    Object.entries(lexerDefinition.modes).forEach(
+      ([currModeName, currModeValue]) => {
+        // Make currModeValue a non-sparse array
+        currModeValue = Array.from(currModeValue)
+        currModeValue.forEach((currTokType, currIdx) => {
```

Comment:
Lodash's `forEach` calls the callback for holes in sparse arrays (with argument `undefined`), whereas `Array.prototype.forEach` skips holes. This was the only place where this difference caused a test to fail, so I inserted the `Array.from` call above to make the array non-sparse. I haven't investigated to see whether the sparse array comes from the test code or whether it's created somewhere in the production code.